In [1]:
import pandas as pd
import matplotlib
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline  

In [2]:
D_stations = pd.read_csv("Datasets/Divvy_Stations_2017_Q1Q2.csv")
D_trips = pd.read_csv("Datasets/Divvy_Trips_2017_Q1.csv")

In [3]:
D_stations.shape

(582, 7)

In [4]:
D_stations.head()

id                               name     city   latitude  longitude  \
0  456                2112 W Peterson Ave  Chicago  41.991178 -87.683593   
1  101                      63rd St Beach  Chicago  41.781016 -87.576120   
2  109                  900 W Harrison St  Chicago  41.874675 -87.650019   
3   21         Aberdeen St & Jackson Blvd  Chicago  41.877726 -87.654787   
4   80  Aberdeen St & Madison (Monroe) St  Chicago  41.881567 -87.655056   

   dpcapacity         online_date  
0          15  2/10/2015 14:04:42  
1          23  7/16/2013 01:27:50  
2          19  7/18/2013 16:45:02  
3          15  6/22/2013 19:07:12  
4          19  6/26/2013 19:00:18

In [5]:
D_trips.shape

(431691, 12)

In [6]:
D_trips.head()

trip_id          start_time            end_time  bikeid  tripduration  \
0  13518905  3/31/2017 23:59:07   4/1/2017 00:13:24    5292           857   
1  13518904  3/31/2017 23:56:25   4/1/2017 00:00:21    4408           236   
2  13518903  3/31/2017 23:55:33   4/1/2017 00:01:21     696           348   
3  13518902  3/31/2017 23:54:46  3/31/2017 23:59:34    4915           288   
4  13518901  3/31/2017 23:53:33   4/1/2017 00:00:28    4247           415   

   from_station_id            from_station_name  to_station_id  \
0               66         Clinton St & Lake St            171   
1              199       Wabash Ave & Grand Ave             26   
2              520   Greenview Ave & Jarvis Ave            432   
3              110        Dearborn St & Erie St            142   
4              327  Sheffield Ave & Webster Ave            331   

                 to_station_name    usertype  gender  birthyear  
0          May St & Cullerton St  Subscriber    Male     1989.0  
1       McClurg Ct & Illinois St  Subscriber    Male     1990.0  
2            Clark St & Lunt Ave  Subscriber  Female     1979.0  
3           McClurg Ct & Erie St  Subscriber    Male     1985.0  
4  Halsted St & Blackhawk St (*)  Subscriber  Female     1989.0

In [7]:
D_trips.dtypes

trip_id                int64
start_time            object
end_time              object
bikeid                 int64
tripduration           int64
from_station_id        int64
from_station_name     object
to_station_id          int64
to_station_name       object
usertype              object
gender                object
birthyear            float64
dtype: object

# distance

Here's a question. What are the common routes with this dataset?

How many possible routes? Triangle number of stations of 582 stations... or
```
n(n+1) / 2
```


In [8]:
D_trips.loc[D_trips['from_station_id'] == D_trips['to_station_id']]

trip_id          start_time            end_time  bikeid  \
50      13518854  3/31/2017 22:57:21  3/31/2017 23:06:40    3022   
113     13518790  3/31/2017 22:12:22  3/31/2017 22:21:50    1799   
115     13518788  3/31/2017 22:11:45  3/31/2017 22:21:25    4569   
116     13518787  3/31/2017 22:11:21  3/31/2017 22:21:50    2411   
132     13518771  3/31/2017 22:02:23  3/31/2017 22:37:31    5453   
135     13518768  3/31/2017 22:01:25  3/31/2017 22:37:31    5003   
254     13518595  3/31/2017 20:49:03  3/31/2017 21:01:49    1579   
392     13518429  3/31/2017 19:57:49  3/31/2017 20:21:46    4267   
432     13518387  3/31/2017 19:46:34  3/31/2017 20:08:23    4090   
457     13518342  3/31/2017 19:40:38  3/31/2017 19:41:51    1168   
464     13518331  3/31/2017 19:39:00  3/31/2017 19:45:26    2831   
579     13518189  3/31/2017 19:15:06  3/31/2017 19:48:56    1590   
633     13518130  3/31/2017 19:05:12  3/31/2017 19:13:29    3307   
756     13517981  3/31/2017 18:43:55  3/31/2017 18:45:21    4010   
784     13517952  3/31/2017 18:39:39  3/31/2017 18:41:07    3307   
792     13517944  3/31/2017 18:38:18  3/31/2017 18:55:12    4690   
828     13517904  3/31/2017 18:33:09  3/31/2017 18:34:47    2323   
1062    13517631  3/31/2017 18:02:36  3/31/2017 18:03:39    2362   
1086    13517604  3/31/2017 17:59:48  3/31/2017 18:12:24     745   
1112    13517575  3/31/2017 17:58:05  3/31/2017 18:52:26    3444   
1181    13517479  3/31/2017 17:49:39  3/31/2017 18:00:21    4907   
1562    13517056  3/31/2017 17:18:39  3/31/2017 17:20:44    1606   
1814    13516793  3/31/2017 16:58:10  3/31/2017 16:59:45    3698   
1968    13516622  3/31/2017 16:45:35  3/31/2017 16:55:08    5404   
2048    13516521  3/31/2017 16:35:19  3/31/2017 16:40:22     481   
2277    13516226  3/31/2017 16:05:54  3/31/2017 16:08:29    4903   
2397    13516071  3/31/2017 15:45:31  3/31/2017 15:49:36    2144   
2401    13516065  3/31/2017 15:45:04  3/31/2017 15:49:46    3942   
2476    13515978  3/31/2017 15:34:13  3/31/2017 17:19:49    5194   
2655    13515762  3/31/2017 14:57:09  3/31/2017 15:27:47    4694   
...          ...                 ...                 ...     ...   
431016  12980023   1/1/2017 12:57:38   1/1/2017 13:28:26    5639   
431163  12979848   1/1/2017 12:16:11   1/1/2017 12:23:53    3558   
431165  12979846   1/1/2017 12:15:57   1/1/2017 12:23:45    3795   
431184  12979821   1/1/2017 12:01:18   1/1/2017 12:11:05    2315   
431192  12979812   1/1/2017 11:58:07   1/1/2017 13:46:49    1913   
431236  12979749   1/1/2017 11:43:48   1/1/2017 12:28:04    5231   
431237  12979748   1/1/2017 11:43:29   1/1/2017 12:28:04    2956   
431239  12979746   1/1/2017 11:43:06   1/1/2017 12:26:43    4460   
431286  12979681   1/1/2017 11:12:52   1/1/2017 11:23:08    3627   
431288  12979679   1/1/2017 11:10:49   1/1/2017 12:15:02    3810   
431289  12979678   1/1/2017 11:10:39   1/1/2017 12:15:02     422   
431290  12979677   1/1/2017 11:10:24   1/1/2017 12:15:02    2461   
431297  12979666   1/1/2017 11:03:06   1/1/2017 11:26:01    5852   
431327  12979617   1/1/2017 10:45:37   1/1/2017 11:00:33    5557   
431330  12979614   1/1/2017 10:44:57   1/1/2017 11:04:38    5656   
431358  12979585   1/1/2017 10:29:13   1/1/2017 10:36:12    4830   
431413  12979518   1/1/2017 09:56:24   1/1/2017 10:22:17     368   
431415  12979515   1/1/2017 09:55:40   1/1/2017 11:32:12    4502   
431418  12979512   1/1/2017 09:55:19   1/1/2017 11:32:20    1272   
431435  12979493   1/1/2017 09:44:04   1/1/2017 10:37:18    3198   
431445  12979482   1/1/2017 09:34:45   1/1/2017 10:15:50    2423   
431451  12979476   1/1/2017 09:28:23   1/1/2017 09:35:33    2352   
431528  12979397   1/1/2017 06:35:07   1/1/2017 07:43:50     524   
431529  12979396   1/1/2017 06:34:35   1/1/2017 07:57:37    2423   
431538  12979386   1/1/2017 05:00:58   1/1/2017 21:43:54    3482   
431544  12979379   1/1/2017 04:04:43   1/1/2017 04:06:00    1432   
431575  12979347   1/1/2017 02:42:35   1/1/2017 02:43:46    2387 

Right, so ~1.8% of riders end up where they started